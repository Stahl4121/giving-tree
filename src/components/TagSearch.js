/* eslint-disable no-use-before-define */
import React from 'react';
import TextField from '@material-ui/core/TextField';
import Autocomplete from '@material-ui/lab/Autocomplete';
import CircularProgress from '@material-ui/core/CircularProgress';
import $ from 'jquery';
import { Button } from '@material-ui/core';

class TagSearch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tags: [{}],
      //tagString: "",
      activeTags: [],
      loading: false
    };

    this.updateRequest = this.updateRequest.bind(this);

  };

  updateRequest() {
		this.props.getTag(this.state);
  }

  handleAutoChange = (event, values) => {
    //this.setState({tagString: values}, this.updateRequest);
    this.setState({
      activeTags: values
    }, () => {
      // This will output an array of objects
      // given by Autocompelte options property.
      console.log(this.state.activeTags);
    });
  }

  handleChange = (event) => {

    this.setState({ loading: true });

    var settings = {
      async: "true",
      crossDomain: "true",
      url: "https://deezerdevs-deezer.p.rapidapi.com/search?q=" + event.target.value,
      method: "GET",
      headers: {
        "x-rapidapi-host": "deezerdevs-deezer.p.rapidapi.com",
        "x-rapidapi-key": "f33e47e69fmshe427476175d1511p18d30djsn436f42242136"
      },
      context: this
    }

    $.ajax(settings).done(function (response) {
      if (typeof response.data !== 'undefined') {
        var newTags = [];
        for (var i = 0; i < response.data.length; i++) {
          newTags.push({ label: response.data[i].title + ' by ' + response.data[i].artist.name });
        }

        this.setState({ tags: newTags, loading: false });
      }
    });

  };

  render() {
    return (
      <Autocomplete
        id="combo-box-demo"
        options={this.state.tags}
        autoComplete
        disableOpenOnFocus
        multiple
        getOptionLabel={option => option.label}
        onChange={this.handleAutoChange}
        style={{ top: "auto", bottom: "auto", height:"auto", postion: "absolute" }}
        renderInput={params => (
             
          <TextField {...params}
            label="Select Tags"
            variant="outlined"
            fullWidth
            //value={this.state.tagString}
            onChange={this.handleChange}
            InputProps={{
              ...params.InputProps,
              endAdornment: (
                <React.Fragment>
                  {this.state.loading ? <CircularProgress color="inherit" size={20} /> : null}
                  {/*params.InputProps.endAdornment*/}
                </React.Fragment>
              ),
            }}
          />
          
        )}
      />
    );
  }
}

export default(TagSearch);