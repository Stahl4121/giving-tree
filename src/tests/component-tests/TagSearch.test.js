import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter, StaticRouter } from 'react-router-dom'

import { TagSearch } from '../../components/TagSearch.js';
/*import firebase, {mockFirebase} from '../__mocks__/firebase';
jest.mock('../__mocks__/firebase');*/

const { mockFirebase } = require('firestore-jest-mock');
const { mockCollection, mockWhere } = require('firestore-jest-mock/mocks/firestore');


describe('Tag Search basics', () => {

  it('Renders correctly', () => {
    const wrap = shallow(<TagSearch />)
    expect(wrap).toMatchSnapshot();
  });
  
})

describe('TagSearch method: UpdateSearch', () => {

  
  mockFirebase({
    database: {
      tags: [
        { tags : ['tagOne', 'tagTwo'] },
      ],
    },
  });

  
  it('Testing the test',  () =>{
    const firebase = require('firebase');
    var db = firebase.firestore();
    var dbRef = db.collection("tags");

    dbRef.get().then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        var dbTags = doc.data().tags;
        expect(dbTags).toEqual(['tagOne', 'tagTwo'])
      });

    });
    
  });


  it('Gets tags from the database',  () =>{
    const firebase = require('firebase');
    const wrap = shallow(<TagSearch.WrappedComponent location={'/'}/>)
    const inst = wrap.instance();
    expect(wrap.find(TagSearch).state('tags')).toBe([{}]);
    inst.updateSearch();
    expect(wrap.find(TagSearch).state('tags')).toBe(['tagOne', 'tagTwo']);
    
  });

})

describe('TagSearch method: ComponentWillMount', () => {

  /*it('No prior location', () => {
    // TODO: No location
  });

  it('Prior location 1', () => {
    // TODO: Prior location
  });

  it('Prior location, returns element', () => {
    // TODO: Prior location
  });*/
  
})

describe('TagSearch method: SetDefaultAuto', () => {

  
  
})

describe('TagSearch method: SetCached', () => {

  
  
})

describe('TagSearch method: RetrieveCache', () => {

  
  
})

describe('TagSearch method: HandleClick', () => {

  
  
})

describe('TagSearch method: UpdateParent', () => {

  
  
})

describe('TagSearch method: HandleAutoChange', () => {

  
  
})
