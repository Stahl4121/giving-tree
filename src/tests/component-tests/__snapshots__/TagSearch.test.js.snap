// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Search basics Renders correctly 1`] = `ShallowWrapper {}`;
